Gems/ROS2/Code/Source/Utilities/ROS2Names.cpp

```diff
@@ -0,0 +1,132 @@
+                input.c_str());
+        }
+
+        const AZStd::string stringToReplaceViolations(1, underscore);
+        const AZStd::regex ros2Disallowedlist("[^0-9|a-z|A-Z|_]");
+        rosified = AZStd::regex_replace(rosified, ros2Disallowedlist, stringToReplaceViolations);
+
+        if (AZStd::isdigit(rosified[0]) || (input[0] != underscore && rosified[0] == underscore))
+        { // Prepend "o3de_" if it would otherwise start with a number (which would violate ros2 name requirements)
+            // Also, starting with '_' is not desired unless explicit. Topics/namespaces/parameters starting with "_" are hidden by default.
+            const AZStd::string prependToNumberStart = "o3de_";
+            rosified = prependToNumberStart + rosified;
+        }
+
+        if (input != rosified)
+        {
+            AZ_TracePrintf(
+                "RosifyName",
+                "Name '%s' has been changed to '%s' to conform with ros2 naming restrictions\n",
+                input.c_str(),
+                rosified.c_str());
+        }
+        return rosified;
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateNamespace(const AZStd::string& ros2Namespace)
+    {
+        auto ros2GlobalizedNamespace = ros2Namespace;
+        const char namespacePrefix = '/';
+        if (!ros2Namespace.starts_with(namespacePrefix))
+        { // Prepend "/" if not included, this is done automatically by rclcpp so "/"-less namespaces are ok.
+            ros2GlobalizedNamespace = namespacePrefix + ros2Namespace;
+        }
+
+        int validationResult = 0;
+        auto ret = rmw_validate_namespace(ros2GlobalizedNamespace.c_str(), &validationResult, NULL);
+        if (ret != RMW_RET_OK)
+        {
+            AZ_Error("ValidateNamespace", false, "Call to rmw validation for namespace failed");
+            return AZ::Failure(AZStd::string("Unable to validate namespace due to rmw error"));
+        }
+
+        if (validationResult != RMW_NAMESPACE_VALID)
+        {
+            return AZ::Failure(AZStd::string(rmw_namespace_validation_result_string(validationResult)));
+        }
+
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateNamespaceField(void* newValue, const AZ::Uuid& valueType)
+    {
+        if (azrtti_typeid<AZStd::string>() != valueType)
+        {
+            return AZ::Failure(AZStd::string("Unexpected field type: the only valid input is a character string"));
+        }
+
+        const AZStd::string& ros2Namespace(*reinterpret_cast<const AZStd::string*>(newValue));
+        return ValidateNamespace(ros2Namespace);
+    }
+
+    AZ::Outcome<void, AZStd::string> ROS2Names::ValidateTopic(const AZStd::string& topic)
+    {
+        int validationResult = 0;
+        size_t invalidIndex; // Unused. We could choose to display it in information, but it is not necessary.
```

Comment:
Should remove the comment and tag with ```[[maybe_unused]]``` instead. This may cause a build failure in gcc
```suggestion
        [[maybe_unused]] size_t invalidIndex;
```

Reply:
done